designer/server/src/main/scala/pl/touk/nussknacker/ui/api/AppApiEndpoints.scala

```diff
@@ -1,150 +1,267 @@
+    final case class HealthCheckProcessErrorResponseDto private(status: HealthCheckProcessErrorResponseDto.Status,
+                                                                message: Option[String],
+                                                                processes: Option[Set[String]])
+    object HealthCheckProcessErrorResponseDto {
 
-    private[AppApiEndpoints] object Codecs {
+      sealed trait Status extends EnumEntry with Uppercase
+      object Status extends Enum[Status] with CirceEnum[Status] {
+        case object Error extends Status
 
-      implicit val healthCheckProcessSuccessResponseDtoCodec: CirceCodec[HealthCheckProcessSuccessResponseDto.type] = {
-        CirceCodec.from(
-          Decoder
-            .forProduct3[HealthCheckProcessSuccessResponseDto.type, String, Option[String], Option[Set[String]]](
-            "status", "message", "processes"
-          ) {
-            case ("OK", None, None) =>
-              HealthCheckProcessSuccessResponseDto
-            case invalid =>
-              throw new IllegalArgumentException(s"Cannot deserialize [$invalid]")
-          },
-          Encoder
-            .forProduct3[HealthCheckProcessSuccessResponseDto.type, String, Option[String], Option[Set[String]]](
-            "status", "message", "processes"
-          )(
-            _ => ("OK", None, None)
-          )
-        )
+        override def values = findValues
       }
 
-      implicit val healthCheckProcessErrorResponseDtoCodec: CirceCodec[HealthCheckProcessErrorResponseDto] = {
-        CirceCodec.from(
-          Decoder
-            .forProduct3[HealthCheckProcessErrorResponseDto, String, Option[String], Option[Set[String]]](
-              "status", "message", "processes"
-            ) {
-              case ("ERROR", message, processes) =>
-                HealthCheckProcessErrorResponseDto(message, processes)
-              case invalid =>
-                throw new IllegalArgumentException(s"Cannot deserialize [$invalid]")
-            },
-          Encoder
-            .forProduct3[HealthCheckProcessErrorResponseDto, String, Option[String], Option[Set[String]]](
-              "status", "message", "processes"
-            )(
-              dto => ("ERROR", dto.message, dto.processes)
-            )
-        )
-      }
+      def apply(message: Option[String],
+                processes: Option[Set[String]]) =
+        new HealthCheckProcessErrorResponseDto(status = Status.Error, message, processes)
+    }
 
-      implicit val buildInfoDtoCodec: CirceCodec[BuildInfoDto] = {
+    @derive(schema)
+    final case class BuildInfoDto(name: String,
+                                  gitCommit: String,
+                                  buildTime: String,
+                                  version: String,
+                                  processingType: Map[String, Map[String, String]],
+                                  globalBuildInfo: Option[Map[String, String]] = None)
```

Comment:
Why it was better to extract these fields into a nested object instead of keeping them in the root level? Am I right that we don't have an option to say to tapir that we have json schema's `additionalProperties`?

Reply:
Having this schema more static IMO is not the goal. The previous approach with inlined additional properties was more idiomatic. It is pretty natural to have json witj some static fields and some other dynamic. The only thing that you have to do is to write `additionalProperties: true` in your schema. Unfortunately, I see that there is no way to modify the generated schema to make it work this way. You can only change some minor details: https://tapir.softwaremill.com/en/latest/endpoint/schemas.html#using-implicits. Here is a similar issue: https://github.com/softwaremill/tapir/issues/1699 . In this situation, it looks like this approach is a reasonable trade-off.